src/utils/ShowModal.ts

```diff
@@ -0,0 +1,57 @@
+export class ShowModal{
+    // this function can show modal view(window) by given raw html template string.
+    // you can also specify the parent container class by passing @context param
+    // @return nothing
+    public static showModal(template:string, context = ''):void {
+        if (!template) {
+            throw new Error('showModal function, no template is given!')
+        }
+        const htmlTemplate = 
+        `<div id='modalwindow' class='b-modal-window-wrapper'>
+            <div id='modalcontent' class='b-modal-window ${context}'>${template}</div>
+        </div>`
+
+        const modalWindow = document.querySelector('#modalwindow');
+        if (modalWindow) {
+            // remove existing modal
+            modalWindow.remove();
+        }
+        // append template to <body />
+        document.body.insertAdjacentHTML('afterBegin', htmlTemplate);
+        const modalWindowNode = document.querySelector('#modalwindow');
```

Comment:
Тут можно так же добавить типы
```suggestion
        const modalWindowNode = document.querySelector<HTMLDivElement>('#modalwindow');
```

Reply:
Сначала не понял, а потом понял :)